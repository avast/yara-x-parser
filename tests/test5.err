SyntaxError("expected expression", 144..144)