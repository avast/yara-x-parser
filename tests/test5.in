//Global comment

//Rule comment
rule test
{
	//Rule block comment

	//String comment
	strings:
		$a = "foo"
		$b = "bar"
	condition:
		$a or
		b and true
}
