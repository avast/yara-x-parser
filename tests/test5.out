SOURCE_FILE@0..157
  COMMENT@0..16 "//Global comment"
  WHITESPACE@16..18 "\n\n"
  RULE@18..156
    COMMENT@18..32 "//Rule comment"
    WHITESPACE@32..33 "\n"
    RULE_KW@33..37 "rule"
    WHITESPACE@37..38 " "
    IDENTIFIER@38..42 "test"
    WHITESPACE@42..43 "\n"
    BLOCK_EXPR@43..156
      L_BRACE@43..44 "{"
      WHITESPACE@44..46 "\n\t"
      COMMENT@46..66 "//Rule block comment"
      WHITESPACE@66..69 "\n\n\t"
      STRINGS@69..121
        COMMENT@69..85 "//String comment"
        WHITESPACE@85..87 "\n\t"
        STRINGS_KW@87..94 "strings"
        COLON@94..95 ":"
        WHITESPACE@95..98 "\n\t\t"
        VARIABLE_STMT@98..108
          VARIABLE@98..100 "$a"
          WHITESPACE@100..101 " "
          ASSIGN@101..102 "="
          WHITESPACE@102..103 " "
          PATTERN@103..108
            STRING_LIT@103..108 "\"foo\""
        WHITESPACE@108..111 "\n\t\t"
        VARIABLE_STMT@111..121
          VARIABLE@111..113 "$b"
          WHITESPACE@113..114 " "
          ASSIGN@114..115 "="
          WHITESPACE@115..116 " "
          PATTERN@116..121
            STRING_LIT@116..121 "\"bar\""
      WHITESPACE@121..123 "\n\t"
      CONDITION@123..154
        CONDITION_KW@123..132 "condition"
        COLON@132..133 ":"
        WHITESPACE@133..136 "\n\t\t"
        BOOLEAN_EXPR@136..154
          BOOLEAN_TERM@136..138
            VARIABLE@136..138 "$a"
          WHITESPACE@138..139 " "
          OR_KW@139..141 "or"
          WHITESPACE@141..144 "\n\t\t"
          BOOLEAN_EXPR@144..154
            BOOLEAN_EXPR@144..154
              BOOLEAN_TERM@144..145
                PRIMARY_EXPR@144..145
                  IDENTIFIER@144..145 "b"
              WHITESPACE@145..146 " "
              AND_KW@146..149 "and"
              WHITESPACE@149..150 " "
              BOOLEAN_EXPR@150..154
                BOOLEAN_TERM@150..154
                  BOOL_LIT@150..154 "true"
      WHITESPACE@154..155 "\n"
      R_BRACE@155..156 "}"
  WHITESPACE@156..157 "\n"
